src/charm.py

```diff
@@ -139,6 +144,12 @@ def __init__(self, *args):
         self.framework.observe(self.on.update_status, self._on_update_status)
         self.framework.observe(self.on.upgrade_charm, self._on_upgrade_charm)
 
+        # Remote configuration events
+        self.framework.observe(
+            self.on[self._remote_configuration_relation_name].relation_changed,
```

Comment:
By convention, the charm should not register relation events, only the lib does that.
Instead, the charm should observe custom lib events.

Reply:
I'm not sure I understand - you're saying that the charm shouldn't react on default `relation_changed` event? Instead I should create a custom lib event which will be emitted when the default `relation_changed` event happens?

Reply:
Iiuc, then yes:
Only the lib registers its own rel's events (on behalf of the charm). All the logic related to rel events is inside the lib anyway. When You need to communicate anything back to the charm you use custom events. This is to encourage reduced coupling (i.e. charm lib should not access charm's attributes or methods).

Again, this is not a restriction, but our convention.

Reply:
Fixed.